Week03/Assignment/ypang_02.py

```diff
@@ -0,0 +1,69 @@
+#!/bin/env python
+
+########################################
+#      Challenge- More functions       #
+########################################
+
+import pandas as pd
+
+
+def one_year_csv_writer(this_year, all_data, directory, root):
+    """
+    Writes a csv file for data from a given year.
+
+    this_year --- year for which data is extracted
+    all_data --- DataFrame with multi-year data
+    """
+    ###### Q4 CHANGE HERE #####
+    print("in one: " + random_variable)
+    #print("in one: " + random_variable2) << This will cause error
+    ###########################
+
+    # Select data for the year
+    surveys_year = all_data[all_data.year == this_year]
+
+    # Write the new DataFrame to a csv file
+    ###### Q1 CHANGE HERE #####
+    filename = directory + "/" + root + str(this_year) + '.csv'
+    ###########################
+    surveys_year.to_csv(filename)
+    ###### Q3 CHANGE HERE #####
+    print(filename + " written.")
+    ###########################
+
+def yearly_data_csv_writer(start_year, end_year, all_data, directory, root):
+    """
+    Writes separate CSV files for each year of data.
+
+    start_year --- the first year of data we want
+    end_year --- the last year of data we want
+    all_data --- DataFrame with multi-year data
```

Comment:
Nice docstrings! If you want to use an existing standard instead of developing your own (I tried this, at first -- it was really laborious!), the [NumPy docstring standards](https://numpydoc.readthedocs.io/en/latest/format.html) work really well, in my opinion.  I've gone to using them in my own code, and all of the projects that I contribute to have it as one possible standard to follow.